Modules/_datetimemodule.c

```diff
@@ -47,39 +48,184 @@ typedef struct {
     PyObject *us_per_week;     // 1e6 * 3600 * 24 * 7 as Python int
     PyObject *seconds_per_day; // 3600 * 24 as Python int
 
-    /* The interned UTC timezone instance */
-    PyObject *utc;
-
     /* The interned Unix epoch datetime instance */
     PyObject *epoch;
-
-    /* While we use a global state, we ensure it's only initialized once */
-    int initialized;
 } datetime_state;
 
-static datetime_state _datetime_global_state;
+#define NO_STATE NULL
```

Comment:
I picked `NO_STATE` because it is used in place of `datetime_state` pointers.  I suppose it could have been `NO_STATE_NEEDED` or `NO_STATE_USED`, but figure `NO_STATE` is clear enough.  If you think it isn't then I'd be glad to sort that out.

Regardless, using `STATIC` would make it less clear, IMHO.  I could see perhaps some value in adding something like a `_STATIC` suffix to the same macros to which we pass `NO_STATE`, but I'm not sure that would pay for itself.